<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>resim</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="margin">
     <number>0</number>
    </property>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Setup</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="1" column="0">
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Devices</string>
          </property>
          <layout class="QGridLayout" name="gridLayout">
           <item row="0" column="0" colspan="3">
            <widget class="QListWidget" name="deviceList"/>
           </item>
           <item row="1" column="0">
            <widget class="QPushButton" name="addDevice">
             <property name="text">
              <string>Add</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QPushButton" name="removeDevice">
             <property name="text">
              <string>Remove</string>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QPushButton" name="deviceSettings">
             <property name="text">
              <string>Settings</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QGroupBox" name="groupBox_3">
          <property name="title">
           <string>CPU</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <widget class="QComboBox" name="cpuModel"/>
           </item>
           <item>
            <widget class="QListWidget" name="initialRegisterList"/>
           </item>
           <item>
            <widget class="QPushButton" name="changeRegister">
             <property name="text">
              <string>Change selected register...</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="1" rowspan="2">
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>Memory Content</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_3">
           <item row="1" column="1">
            <widget class="QPushButton" name="removeMemoryContent">
             <property name="text">
              <string>Remove</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QPushButton" name="addMemoryContent">
             <property name="text">
              <string>Add</string>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QPushButton" name="changeMemoryContent">
             <property name="text">
              <string>Settings</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0" colspan="3">
            <widget class="QListWidget" name="memoryContentList"/>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>CPU State</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="0" column="1">
         <widget class="QGroupBox" name="groupBox_4">
          <property name="title">
           <string>Registers</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QListWidget" name="registerState"/>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QGroupBox" name="groupBox_5">
          <property name="title">
           <string>General</string>
          </property>
          <layout class="QFormLayout" name="formLayout">
           <item row="0" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>State:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="cpuState">
             <property name="text">
              <string>Running</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Time:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLabel" name="time">
             <property name="text">
              <string>0</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="0" colspan="2">
         <widget class="QPushButton" name="refreshProcessorState">
          <property name="text">
           <string>Refresh</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_4">
       <attribute name="title">
        <string>Memory State</string>
       </attribute>
      </widget>
      <widget class="QWidget" name="tab_5">
       <attribute name="title">
        <string>Devices</string>
       </attribute>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>Log</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="2" column="2">
         <widget class="QPushButton" name="saveLog">
          <property name="text">
           <string>Save</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QPushButton" name="refreshLog">
          <property name="text">
           <string>Refresh</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QPushButton" name="clearLog">
          <property name="text">
           <string>Clear</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0" colspan="3">
         <widget class="QListWidget" name="log"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
    <addaction name="actionRecentFiles"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionStartSimulation"/>
    <addaction name="actionPauseSimulation"/>
    <addaction name="actionStopSimulation"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew"/>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
  </widget>
  <widget class="QToolBar" name="toolBar_2">
   <property name="windowTitle">
    <string>toolBar_2</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionStopSimulation"/>
   <addaction name="actionPauseSimulation"/>
   <addaction name="actionStartSimulation"/>
  </widget>
  <action name="actionAbout">
   <property name="icon">
    <iconset resource="../resim.qrc">
     <normaloff>:/icons/icons/help-about.png</normaloff>:/icons/icons/help-about.png</iconset>
   </property>
   <property name="text">
    <string>About...</string>
   </property>
  </action>
  <action name="actionStartSimulation">
   <property name="icon">
    <iconset resource="../resim.qrc">
     <normaloff>:/icons/icons/media-playback-start.png</normaloff>:/icons/icons/media-playback-start.png</iconset>
   </property>
   <property name="text">
    <string>Start simulation</string>
   </property>
  </action>
  <action name="actionPauseSimulation">
   <property name="icon">
    <iconset resource="../resim.qrc">
     <normaloff>:/icons/icons/media-playback-pause.png</normaloff>:/icons/icons/media-playback-pause.png</iconset>
   </property>
   <property name="text">
    <string>Pause simulation</string>
   </property>
  </action>
  <action name="actionStopSimulation">
   <property name="icon">
    <iconset resource="../resim.qrc">
     <normaloff>:/icons/icons/media-playback-stop.png</normaloff>:/icons/icons/media-playback-stop.png</iconset>
   </property>
   <property name="text">
    <string>Stop simulation</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="icon">
    <iconset resource="../resim.qrc">
     <normaloff>:/icons/icons/document-new.png</normaloff>:/icons/icons/document-new.png</iconset>
   </property>
   <property name="text">
    <string>New...</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="../resim.qrc">
     <normaloff>:/icons/icons/document-open.png</normaloff>:/icons/icons/document-open.png</iconset>
   </property>
   <property name="text">
    <string>Open...</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="../resim.qrc">
     <normaloff>:/icons/icons/document-save.png</normaloff>:/icons/icons/document-save.png</iconset>
   </property>
   <property name="text">
    <string>Save...</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="icon">
    <iconset resource="../resim.qrc">
     <normaloff>:/icons/icons/application-exit.png</normaloff>:/icons/icons/application-exit.png</iconset>
   </property>
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="actionRecentFiles">
   <property name="icon">
    <iconset resource="../resim.qrc">
     <normaloff>:/icons/icons/document-open-recent.png</normaloff>:/icons/icons/document-open-recent.png</iconset>
   </property>
   <property name="text">
    <string>Recent files</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="icon">
    <iconset resource="../resim.qrc">
     <normaloff>:/icons/icons/document-save-as.png</normaloff>:/icons/icons/document-save-as.png</iconset>
   </property>
   <property name="text">
    <string>Save as...</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="../resim.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionAbout</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>about()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNew</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>newConfig()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>open()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>save()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSaveAs</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cpuModel</sender>
   <signal>activated(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>selectProcessor(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>254</x>
     <y>137</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>changeRegister</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>changeRegister()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>254</x>
     <y>318</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addDevice</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>addDevice()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>101</x>
     <y>536</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>removeDevice</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>removeDevice()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>254</x>
     <y>536</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>deviceSettings</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>changeDevice()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>407</x>
     <y>536</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addMemoryContent</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>addMemoryContent()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>555</x>
     <y>542</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>removeMemoryContent</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>removeMemoryContent()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>646</x>
     <y>542</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>changeMemoryContent</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>changeMemoryContent()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>737</x>
     <y>542</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>refreshLog</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>refreshLog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>138</x>
     <y>553</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clearLog</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>clearLog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>399</x>
     <y>553</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveLog</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveLog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>661</x>
     <y>553</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>refreshProcessorState</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>refreshProcessorState()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>399</x>
     <y>553</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionStartSimulation</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>start()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPauseSimulation</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>pause()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionStopSimulation</sender>
   <signal>activated()</signal>
   <receiver>MainWindow</receiver>
   <slot>stop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>about()</slot>
  <slot>newConfig()</slot>
  <slot>open()</slot>
  <slot>save()</slot>
  <slot>saveAs()</slot>
  <slot>selectProcessor(int)</slot>
  <slot>changeRegister()</slot>
  <slot>addDevice()</slot>
  <slot>removeDevice()</slot>
  <slot>changeDevice()</slot>
  <slot>addMemoryContent()</slot>
  <slot>removeMemoryContent()</slot>
  <slot>changeMemoryContent()</slot>
  <slot>refreshLog()</slot>
  <slot>clearLog()</slot>
  <slot>saveLog()</slot>
  <slot>refreshProcessorState()</slot>
  <slot>start()</slot>
  <slot>pause()</slot>
  <slot>stop()</slot>
 </slots>
</ui>
